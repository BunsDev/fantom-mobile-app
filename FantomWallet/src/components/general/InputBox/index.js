// @flow
import React from 'react';
import { TextInput, View, Text } from 'react-native';
import styles from './styles';

type Props = {
  phraseNumber: number | string,
  text: string,
  onChangeText: string => void,
};

/**
 * InputBox: A component for rendering input box in Captcha Verification Screen.
 * @param  {} {phraseNumber - Contains phrase number to be displayed on input box.
 * @param  {} text - To Set Text to be displayed in input box.
 * @param  {Props} onChangeText} - Callback function to handle onChange on inputbox.
 */
export default ({ phraseNumber, text, onChangeText }: Props) => (
  <View>
    <Text style={styles.phraseNumber}> {phraseNumber}</Text>
    <TextInput
      onChangeText={onChangeText}
      value={text}
      style={styles.textBox}
      autoCapitalize="none"
      underlineColorAndroid="transparent"
      autoCorrect={false}
    />
  </View>
);
