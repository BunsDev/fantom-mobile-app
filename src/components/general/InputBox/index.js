// @flow
import React from 'react';
import { TextInput, View, Text } from 'react-native';
import styles from './styles';

/**
 * InputBox: A component for rendering input box in Captcha Verification Screen.
 * @param  {} {phraseNumber - Contains phrase number to be displayed on input box.
 * @param  {} text - To Set Text to be displayed in input box.
 * @param  {Props} onChangeText} - Callback function to handle onChange on inputbox.
 */
export default (props: TInputBoxTypes) => {
  const { phraseNumber, text, onChangeText }=props
  return (
    <View>
      <Text style={styles.phraseNumber}> {phraseNumber}</Text>
      <TextInput
        onChangeText={onChangeText}
        value={text}
        style={styles.textBox}
        autoCapitalize="none"
        underlineColorAndroid="transparent"
        autoCorrect={false}
      />
    </View>
  );
}