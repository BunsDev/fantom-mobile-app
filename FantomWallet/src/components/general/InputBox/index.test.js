import React from 'react';
import { shallow } from 'enzyme';

import InputBox from './index';

test('renders correctly', () => {
  const wrapper = shallow(
    <InputBox phraseNumber={2} text="first second" onChangeText={jest.fn()} />
  );
  expect(wrapper).toMatchSnapshot();
});
