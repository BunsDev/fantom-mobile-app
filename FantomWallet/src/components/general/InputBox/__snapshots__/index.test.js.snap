// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "black",
        "marginBottom": 13.34,
      }
    }
  >
     
    2
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[MockFunction]}
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "rgb(223,224,224)",
        "borderWidth": 1,
        "fontFamily": "SegoeUI-SemiBold",
        "height": 44,
        "padding": 10,
        "width": 675,
      }
    }
    underlineColorAndroid="transparent"
    value="first second"
  />
</View>
`;
